### PR TITLE
Adoption and Interest forms ported to React

### PR DESCRIPTION
**Pull request by RoyEJohnson**
Issue #1789, which was the year of the French Revolution

### CODE REVIEW

File: src/app/pages/interest/interest.js [hunk 1]
```diff
@@ -1,178 +1,189 @@
...
+    const formRef = useRef();
+
+    function validatePage(page) {
+        const validateContactInfo = contactValidatorRef.current;
+
+        if (page === 1 && !validateContactInfo()) {
+            return false;
+        }
+        return true;
     }
```
Comment (nathanstitt) [inline src/app/pages/interest/interest.js:103]: a quibble, but you could: `return Boolean(page !== 1 && validateContactInfo())`

File: src/app/pages/interest/interest.js [hunk 2]
```diff
@@ -1,178 +1,189 @@
...
-            return contactForm.checkSchoolName() || contactForm.validate();
-        };
-        return result;
+        }
+
+        formRef.current = form;
+        if (iframe) {
+            iframe.addEventListener('load', submitAfterDelay);
+        }
+        submitNextBook();
```
Comment (nathanstitt) [inline src/app/pages/interest/interest.js:135]: There's the possibility of a bug here if the form is unmounted in the 300ms after `submitAfterDelay` is called. I think you can work around that by checking if one of the references are still valid in `submitNextBook`.  I think that references are reset to null when unmounting, but not entirely sure about that.  

To test you could increase the timeout to a large number like 10 seconds, submit and then hit browser back button, then watch the console to see what happens.

File: src/app/components/form-input/form-input.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        return () => shellBus.emit('no-sticky');
+    }, []);
+
+    return (
+        <div className="suggestions">
+            <div className="suggestion-box">
+                {
+                    !exactMatch && !accepted && matches.map((match, i) =>
+                        <SuggestionItem value={match} accept={accept} index={i}
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/components/form-input/form-input.jsx:59]: Clearly something I need to pay more attention to.

File: src/app/components/jsx-helpers/jsx-helpers.jsx [hunk 4]
```diff
@@ -13,7 +14,7 @@ export function WindowContextProvider({children}) {
     const [value, setValue] = useState(getValuesFromWindow());
 
     useLayoutEffect(() => {
-        const handleScroll = throttle(() => setValue(getValuesFromWindow()), 40);
+        const handleScroll = () => setValue(getValuesFromWindow());
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/components/jsx-helpers/jsx-helpers.jsx:18]: No, this is not a good situation.

File: src/app/pages/interest/interest.js (hunk 1 above)
Comment (RoyEJohnson) [reply src/app/pages/interest/interest.js:103]: with ||, yes.

File: src/app/pages/interest/interest.js (hunk 2 above)
Comment (RoyEJohnson) [reply src/app/pages/interest/interest.js:135]: It looks like `selectedBooks` got cleared when the form unmounted, so the queue became empty. No complaints in the Console, at least.

**Pull request merged**